pwnlib/data/templates/pwnup.mako

```diff
@@ -161,6 +160,8 @@ continue
 
 io = start()
 
+if args.GDB:
+    log.info(io.recvline())
```

Comment:
This should be added in `pwnlib.gdb`, not to the template, probably right after this:

https://github.com/Gallopsled/pwntools/blob/20cb04984ae9687de3f192c76e1cfa405b00665b/pwnlib/gdb.py#L782-L783

The issue is that doesn't actually happen all of the time -- only for certain (newer?) versions of `gdbserver`.  We can have a `.recvline(timeout=0.5)` and then check to see if the line starts how we expect.  If not, we'll need to `.unrecv(...)` the data to put it back.

Reply:
I see there's already been done some handling for this in `gdb.debug`:
https://github.com/Gallopsled/pwntools/blob/20cb04984ae9687de3f192c76e1cfa405b00665b/pwnlib/gdb.py#L465-L469

Reply:
When running locally, `gdb.debug` spawns `gdbserver` and then invokes `gdb.attach`.

Reply:
Yep.

I've adjusted the code in `gdb.debug`. It should handle gdb's message correctly now.